### PR TITLE
Better foot plant

### PR DESCRIPTION
**Pull request by Stermere**
Makes foot plant work when foot trackers are being used. It's based on https://github.com/SlimeVR/SlimeVR-Server/pull/647

### CODE REVIEW

**Comment by ImUrX** [comment]
Ready for review & merge!

File: server/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweaks.java [hunk 1]
```diff
@@ -837,6 +839,12 @@ private void correctFootRotations() {
 			weightL = getFootPlantWeight(leftFootPosition);
 			weightR = getFootPlantWeight(rightFootPosition);
 
+			// if foot trackers exist add to the weights
+			if (footTrackers) {
+				weightL = weightL * getRotationalDistanceToPlant(leftFootRotation);
+				weightR = weightR * getRotationalDistanceToPlant(rightFootRotation);
```
Comment (ButterscotchV) [inline server/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweaks.java:845]:
```suggestion
				weightL *= getRotationalDistanceToPlant(leftFootRotation);
				weightR *= getRotationalDistanceToPlant(rightFootRotation);
```

**Review by ButterscotchV — APPROVED** [review]

File: server/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweaks.java [hunk 2]
```diff
@@ -810,9 +812,9 @@ private void correctFootRotations() {
 		if (bufferHead == null || bufferHead.getParent() == null)
 			return;
 
-		// if there is a foot tracker for a foot don't correct it
-		if (skeleton.leftFootTracker != null || skeleton.rightFootTracker != null)
-			return;
+		// boolean for if there is a foot tracker
+		boolean footTrackers = skeleton.leftFootTracker != null
+			|| skeleton.rightFootTracker != null;
```
Comment (Louka3000) [inline server/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweaks.java:817]: Could you make it 2 separate variables, one per foot, in order to support any ankle + foot setup?

Comment (Stermere) [reply server/src/main/java/dev/slimevr/tracking/processor/skeleton/LegTweaks.java:817]: for sure! while I'm at it all make toe snap also do that also

**Review by ButterscotchV — APPROVED** [review]

**Review by Louka3000 — APPROVED** [review]

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**